doc/meson.build

```diff
@@ -0,0 +1,132 @@
+private_headers = [
+    'include/vips/almostdeprecated.h',
+    'include/vips/deprecated.h',
+    'include/vips/vips7compat.h',
+    'include/vips/dispatch.h',
+    'include/vips/enumtypes.h',
+    'include/vips/internal.h',
+    'include/vips/thread.h',
+    'include/vips/intl.h',
+    'include/vips/format.h',
+    'include/vips/mask.h',
+    'include/vips/private.h',
+    'include/vips/video.h',
+# ignore all .h files in libvips/*, these are internal
+    'arithmetic/binary.h',
+    'arithmetic/hough.h',
+    'arithmetic/nary.h',
+    'arithmetic/parithmetic.h',
+    'arithmetic/statistic.h',
+    'arithmetic/unary.h',
+    'arithmetic/unaryconst.h',
+    'colour/pcolour.h',
+    'colour/profiles.h',
+    'conversion/bandary.h',
+    'conversion/pconversion.h',
+    'convolution/correlation.h',
+    'convolution/pconvolution.h',
+    'create/pcreate.h',
+    'create/pmask.h',
+    'create/point.h',
+    'draw/drawink.h',
+    'draw/pdraw.h',
+    'foreign/dbh.h',
+    'foreign/jpeg.h',
+    'foreign/magick.h',
+    'foreign/pforeign.h',
+    'foreign/tiff.h',
+    'freqfilt/pfreqfilt.h',
+    'histogram/hist_unary.h',
+    'histogram/phistogram.h',
+    'iofuncs/sink.h',
+    'iofuncs/vipsmarshal.h',
+    'morphology/pmorphology.h',
+    'mosaicing/global_balance.h',
+    'mosaicing/pmosaicing.h',
+    'resample/presample.h',
+    'resample/templates.h'
+]
+
+markdown_content_files = files(
+    'How-it-works.md',
+    'libvips-from-C++.md',
+    'Using-vipsthumbnail.md',
+    'How-it-opens-files.md',
+    'Examples.md',
+    'Cite.md',
+    'binding.md',
+    'Making-image-pyramids.md',
+)
+
+markdown_content_files_docbook = []
+foreach markdown_content_file : markdown_content_files
+    # we have some files in markdown ... convert to docbook for gtk-doc
+    # pandoc makes section headers, we want refsect3 for gtk-doc
+    markdown_content_files_docbook += configure_file(
+        input: markdown_content_file,
+        command: [
+            find_program('pandoc'),
```

Comment:
I think you probably want to put that in the same get_option() guard as your `gnome.gtkdoc()` invocation.

With the current implementation, you always build this pandoc output, which also means you always require pandoc installed as a build dependency, even when configuring with gtk_doc disabled.

Reply:
Also slightly interested to see you using configure_file there... this triggers buildsystem regeneration when the inputs are modified. Although maybe they aren't frequently modified...

Reply:
Ah right, I've added a guard around the folder inclusion then